service/raft/raft_address_map.hh

```diff
@@ -205,12 +291,22 @@ public:
     // This means that we cannot remap the entry's actual inet_address but
     // nonetheless the function can be used to promote the entry from
     // expiring to permanent or vice versa.
+    //
+    // Non-expiring entries can only be set on shard 0 and are replicated to other shards.
+    // Expiring entries are shard-local.
```

Comment:
Same comment - it's easier to not care about these rules in the address map and not have to explain them.

Reply:
This is an issue of correctness: see my comment below.